jbpm-case-mgmt/jbpm-case-mgmt-impl/src/test/java/org/jbpm/casemgmt/impl/SubCaseServiceImplTest.java

```diff
@@ -485,4 +486,60 @@ public void testStartCaseWithSubCaseAbortProcessInstanceOfSubCase() {
         }
     }
 
+    @Test
+    public void testCaseWithSubCaseQuery() {
+
+        Map<String, OrganizationalEntity> roleAssignments = new HashMap<>();
+        roleAssignments.put("owner", new UserImpl("john"));
+        roleAssignments.put("manager", new UserImpl("mary"));
+
+        Map<String, Object> data = new HashMap<>();
+        data.put("name", "John Doe");
+        CaseFileInstance caseFile = caseService.newCaseFileInstance(deploymentUnit.getIdentifier(), BASIC_SUB_CASE_P_ID, data, roleAssignments);
+
+        String caseId = caseService.startCase(deploymentUnit.getIdentifier(), BASIC_SUB_CASE_P_ID, caseFile);
+        assertNotNull(caseId);
+        assertEquals(SUB_CASE_ID, caseId);
+        try {
+            CaseInstance cInstance = caseService.getCaseInstance(caseId);
+            assertNotNull(cInstance);
+            assertEquals(deploymentUnit.getIdentifier(), cInstance.getDeploymentId());
+
+            caseService.triggerAdHocFragment(caseId, "Sub Case", null);
+
+            Collection<CaseInstance> caseInstances = caseRuntimeDataService.getCaseInstances(new QueryContext());
+            assertNotNull(caseInstances);
+            assertEquals(2, caseInstances.size());
+
+            // we check the parent
+            List<CaseStatus> allStatus = Arrays.asList(CaseStatus.values());
+            Collection<CaseInstance> subcases = caseRuntimeDataService.getSubCaseInstancesByParentCaseId(SUB_CASE_ID, allStatus, new QueryContext());
+            assertNotNull(subcases);
+            assertEquals(1, subcases.size());
+            List<String> childId = subcases.stream().map(CaseInstance::getCaseId).collect(Collectors.toList());
+            assertEquals(1, childId.size()); // only one child
+            assertTrue(childId.contains(HR_CASE_ID));
+            assertEquals(SUB_CASE_ID, subcases.iterator().next().getParentCaseId());
+
+            Collection<CaseInstance> allSubcases = caseRuntimeDataService.getAllDescendantSubCaseInstancesByParentCaseId(SUB_CASE_ID, allStatus, new QueryContext());
```

Comment:
This test covers just one level (same result as the other API).
To add more levels, it's needed to add the custom task (subcase) inside UserTaskCase.bpmn, and then invoke it as many times as nested cases we want to have:
`caseService.triggerAdHocFragment("HR-000000000..", "New Nested Sub Case", null);`

Reply:
done...